data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbClientWrapper.java

```diff
@@ -46,52 +52,79 @@ public class DynamoDbClientWrapper {
+        if (!dynamoStoreSettings.skipTableCreation()) {
+            try {
+                final CreateTableEnhancedRequest createTableEnhancedRequest = CreateTableEnhancedRequest.builder()
+                        .provisionedThroughput(provisionedThroughput)
+                        .globalSecondaryIndices(EnhancedGlobalSecondaryIndex.builder()
+                                .indexName(SOURCE_STATUS_COMBINATION_KEY_GLOBAL_SECONDARY_INDEX)
+                                .provisionedThroughput(provisionedThroughput)
+                                .projection(Projection.builder().projectionType(ProjectionType.ALL).build())
+                                .build())
+                        .build();
+                table.createTable(createTableEnhancedRequest);
+            } catch (final ResourceInUseException e) {
+                LOG.info("The table creation for {} was already triggered by another instance of data prepper", dynamoStoreSettings.getTableName());
+            }
         }
 
         try(final DynamoDbWaiter dynamoDbWaiter = DynamoDbWaiter.create()) {
-            final DescribeTableRequest describeTableRequest = DescribeTableRequest.builder().tableName(tableName).build();
+            final DescribeTableRequest describeTableRequest = DescribeTableRequest.builder().tableName(dynamoStoreSettings.getTableName()).build();
             final ResponseOrException<DescribeTableResponse> response = dynamoDbWaiter
                     .waitUntilTableExists(describeTableRequest)
                     .matched();
 
             final DescribeTableResponse describeTableResponse = response.response().orElseThrow(
-                    () -> new RuntimeException(String.format("Table %s was not created successfully", tableName))
+                    () -> new RuntimeException(String.format("DynamoDb Table %s could not be found.", dynamoStoreSettings.getTableName()))
             );
 
-            LOG.info("DynamoDB table {} was created successfully for source coordination", describeTableResponse.table().tableName());
+            LOG.debug("DynamoDB table {} was created successfully for source coordination", describeTableResponse.table().tableName());
+
+            if (Objects.nonNull(dynamoStoreSettings.getTtl())) {
+                if (!dynamoStoreSettings.skipTableCreation()) {
+                    dynamoDbClient.updateTimeToLive(UpdateTimeToLiveRequest.builder()
+                            .tableName(table.tableName())
+                            .timeToLiveSpecification(TimeToLiveSpecification.builder()
+                                    .attributeName(TTL_ATTRIBUTE_NAME)
+                                    .enabled(Objects.nonNull(dynamoStoreSettings.getTtl()))
+                                    .build())
+                            .build());
+                }
+
+                final DescribeTimeToLiveResponse describeTimeToLiveResponse = dynamoDbClient.describeTimeToLive(DescribeTimeToLiveRequest.builder()
```

Comment:
Should this be in an else block? It seems you don't need this if the table creation is provided by Data Prepper.

Reply:
No. Because if `ttl` is specified in the config, then we will still use it, meaning ttl needs to configured correctly on the table. So there would be an error on startup saying that you don't have ttl configured on the store correctly, and you can either add it to the store manually, not use ttl, or enable `skip_table_creation` and rerun

Reply:
My point is that I think you don't need to perform this check if `updateTimeToLive` runs. Right? It's not a big deal, but it seems an unnecessary describe request.

Reply:
Oh I see. You're right I think it is unnecessary. Will add the else block for describe call